pkg/driver/distro/distro.go

```diff
@@ -208,7 +220,12 @@ func Build(ctx context.Context,
 	// try to load kernel headers urls from driverkit.
 	if _, found := env[drivertype.KernelDirEnv]; !found {
 		printer.Logger.Debug("Trying to automatically fetch kernel headers.")
-		kernelHeadersPath, cleaner, err := loadKernelHeadersFromDk(d.String(), kr)
+		// KernelVersion needs to be fixed up; it is only used by driverkit Ubuntu builder
+		// and we must ensure that it is correctly set.
+		fixedKr := d.FixupKernel(kr)
+		kVerFixedKr := kr
```

Comment:
Don't overwrite `kr` here, instead use a copy.

Reply:
While this is a bit ugly rn, it will be much better once https://github.com/falcosecurity/falcoctl/pull/356 lands.